x/recovery/keeper/ibc_callbacks_integration_test.go

```diff
@@ -0,0 +1,435 @@
+	suite.Require().NoError(err)
+	err = suite.EvmosChain.App.(*app.Evmos).BankKeeper.SendCoinsFromModuleToAccount(suite.EvmosChain.GetContext(), inflationtypes.ModuleName, suite.IBCOsmosisChain.SenderAccount.GetAddress(), coins)
+	suite.Require().NoError(err)
+
+	// Mint coins on the osmosis side which we'll use to unlock our aevmos
+	coins = sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(10)))
+	err = suite.IBCOsmosisChain.GetSimApp().BankKeeper.MintCoins(suite.IBCOsmosisChain.GetContext(), minttypes.ModuleName, coins)
+	suite.Require().NoError(err)
+	err = suite.IBCOsmosisChain.GetSimApp().BankKeeper.SendCoinsFromModuleToAccount(suite.IBCOsmosisChain.GetContext(), minttypes.ModuleName, suite.IBCOsmosisChain.SenderAccount.GetAddress(), coins)
+	suite.Require().NoError(err)
+
+	// Mint coins on the cosmos side which we'll use to unlock our aevmos
+	coins = sdk.NewCoins(sdk.NewCoin("uatom", sdk.NewInt(10)))
+	err = suite.IBCCosmosChain.GetSimApp().BankKeeper.MintCoins(suite.IBCCosmosChain.GetContext(), minttypes.ModuleName, coins)
+	suite.Require().NoError(err)
+	err = suite.IBCCosmosChain.GetSimApp().BankKeeper.SendCoinsFromModuleToAccount(suite.IBCCosmosChain.GetContext(), minttypes.ModuleName, suite.IBCCosmosChain.SenderAccount.GetAddress(), coins)
+	suite.Require().NoError(err)
+
+	claimparams := claimtypes.DefaultParams()
+	claimparams.AirdropStartTime = suite.EvmosChain.GetContext().BlockTime()
+	claimparams.EnableClaims = true
+	suite.EvmosChain.App.(*app.Evmos).ClaimsKeeper.SetParams(suite.EvmosChain.GetContext(), claimparams)
+
+	params := types.DefaultParams()
+	params.EnableRecovery = true
+	suite.EvmosChain.App.(*app.Evmos).RecoveryKeeper.SetParams(suite.EvmosChain.GetContext(), params)
+
+	suite.pathOsmosisEvmos = ibctesting.NewTransferPath(suite.IBCOsmosisChain, suite.EvmosChain) // clientID, connectionID, channelID empty
+	suite.pathCosmosEvmos = ibctesting.NewTransferPath(suite.IBCCosmosChain, suite.EvmosChain)
+	suite.pathOsmosisCosmos = ibctesting.NewTransferPath(suite.IBCCosmosChain, suite.IBCOsmosisChain)
+	suite.coordinator.Setup(suite.pathOsmosisEvmos) // clientID, connectionID, channelID filled
+	suite.coordinator.Setup(suite.pathCosmosEvmos)
+	suite.coordinator.Setup(suite.pathOsmosisCosmos)
+	suite.Require().Equal("07-tendermint-0", suite.pathOsmosisEvmos.EndpointA.ClientID)
+	suite.Require().Equal("connection-0", suite.pathOsmosisEvmos.EndpointA.ConnectionID)
+	suite.Require().Equal("channel-0", suite.pathOsmosisEvmos.EndpointA.ChannelID)
+}
+
+func TestIBCTestingSuite(t *testing.T) {
+	suite.Run(t, new(IBCTestingSuite))
+}
+
+var (
+	timeoutHeight = clienttypes.NewHeight(1000, 1000)
+
+	uosmoDenomtrace = transfertypes.DenomTrace{
+		Path:      "transfer/channel-0",
+		BaseDenom: "uosmo",
+	}
+	uosmoIbcdenom = uosmoDenomtrace.IBCDenom()
+
+	uatomDenomtrace = transfertypes.DenomTrace{
+		Path:      "transfer/channel-1",
+		BaseDenom: "uatom",
+	}
+	uatomIbcdenom = uatomDenomtrace.IBCDenom()
```

Comment:
do we need to create a denom for the atom -> osmo -> evmos `"uatom"` to test the case?

Reply:
added the denom